server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -710,6 +715,15 @@ public void compactionFailed(TInfo tinfo, TCredentials credentials, String exter
     compactionFailed(Map.of(ecid, KeyExtent.fromThrift(extent)));
   }
 
+  void compactionsFailed(Map<ExternalCompactionId,Pair<KeyExtent,DataLevel>> compactionsByLevel) {
+    // Need to process each level by itself because the conditional tablet mutator does not support
+    // mutating multiple data levels at the same time
+    compactionsByLevel.entrySet().stream()
+        .collect(groupingBy(entry -> entry.getValue().getSecond(),
+            Collectors.toMap(Entry::getKey, e -> e.getValue().getFirst())))
+        .forEach((level, compactions) -> compactionFailed(compactions));
+  }
+
```

Comment:
Could avoid tracking the level in the outer code by computing it here when partitioning.  Would need to rename the function that takes a map of data for a level.  Renamed it to `compactionFailedForLevel` in the suggestion.

```suggestion
  void compactionsFailed(Map<ExternalCompactionId,KeyExtent> compactionsByLevel) {
    // Need to process each level by itself because the conditional tablet mutator does not support
    // mutating multiple data levels at the same time
    compactionsByLevel.entrySet().stream()
        .collect(groupingBy(entry -> DataLevel.of(entry.getValue().tableId()),
            Collectors.toMap(Entry::getKey, Entry::getValue)))
        .forEach((level, compactions) -> compactionFailedForLevel(compactions));
  }
```

Reply:
Thanks for the suggestion, that should simplify things. I can update that and merge later this week when I get some time, probably Wednesday.